incubator/hnc/internal/reconcilers/hnc_config.go

```diff
@@ -60,8 +65,11 @@ type ConfigReconciler struct {
 	enqueueReasonsLock sync.Mutex
 }
 
+// grSet keeps track of a group of unique GRs and their mappings to GVKs.
```

Comment:
I'm thinking something like this might be a bit more clear:

```
type gvkMode struct {
   gvk schema.GroupVersionKind
   mode api.Mode
}

type grSettings map[schema.GroupResource]gvkMode
```

Now, as you go through the `config.Spec.Types`, you're populating a `grSettings` object, and if it's already populated (with a different value), then you know you have a conflict. Make sense?

Reply:
sgtm. Just confirming we will still keep the gvk2gr mapping for `setTypeStatus` from forest to hncconfig status, right?

Reply:
Ah yes, that makes sense.